source/event_stream_channel_handler.c

```diff
@@ -0,0 +1,535 @@
+                    "id=%p: Parsing the message failed with error %s.",
+                    (void *)handler,
+                    aws_error_debug_str(error_code));
+                goto finished;
+            }
+
+            size_t message_size = event_stream_handler->message_buf.len;
+            AWS_LOGF_TRACE(
+                AWS_LS_EVENT_STREAM_CHANNEL_HANDLER, "id=%p: Invoking on_message_received callback.", (void *)handler);
+            event_stream_handler->on_message_received(
+                &received_message, AWS_ERROR_SUCCESS, event_stream_handler->user_data);
+            aws_event_stream_message_clean_up(&received_message);
+            event_stream_handler->current_message_len = 0;
+            event_stream_handler->running_crc = 0;
+            aws_byte_buf_reset(&event_stream_handler->message_buf, true);
+
+            if (!event_stream_handler->manual_window_management) {
+                aws_channel_slot_increment_read_window(slot, message_size);
+            }
+        }
+    }
+
+finished:
+    if (error_code) {
+        event_stream_handler->on_message_received(NULL, error_code, event_stream_handler->user_data);
+        aws_channel_shutdown(slot->channel, error_code);
+    }
+    aws_mem_release(message->allocator, message);
+    return AWS_OP_SUCCESS;
+}
+
+struct message_write_data {
+    struct aws_allocator *allocator;
+    struct aws_channel_task task;
+    struct aws_event_stream_channel_handler *handler;
+    struct aws_event_stream_message *message;
+    aws_event_stream_channel_handler_on_message_written_fn *on_message_written;
+    void *user_data;
+};
+
+static void s_on_message_write_completed_fn(
+    struct aws_channel *channel,
+    struct aws_io_message *message,
+    int err_code,
+    void *user_data) {
+    (void)channel;
+    (void)message;
+
+    struct message_write_data *message_data = user_data;
+    AWS_LOGF_TRACE(
+        AWS_LS_EVENT_STREAM_CHANNEL_HANDLER,
+        "channel=%p: Message write completed. Invoking "
+        "on_message_written callback.",
+        (void *)channel);
+    message_data->on_message_written(message_data->message, err_code, message_data->user_data);
+    aws_mem_release(message_data->allocator, message_data);
+}
+
+static void s_write_handler_message(struct aws_channel_task *task, void *arg, enum aws_task_status status) {
+    (void)task;
+
+    AWS_LOGF_TRACE(AWS_LS_EVENT_STREAM_CHANNEL_HANDLER, "static: Write message task invoked.");
+    struct message_write_data *message_data = arg;
+    if (status == AWS_TASK_STATUS_RUN_READY) {
+        struct aws_event_stream_message *message = message_data->message;
+        struct aws_event_stream_channel_handler *handler = message_data->handler;
+
+        struct aws_byte_cursor message_cur = aws_byte_cursor_from_array(
+            aws_event_stream_message_buffer(message), aws_event_stream_message_total_length(message));
+
+        while (message_cur.len) {
+            AWS_LOGF_TRACE(
```

Comment:
Feels like all the trace messages in here should have the channel/handler pointer as the id?

Reply:
updated all messages in this callback to include id.